jobs/update-colin-filings/update_colin_filings.py

```diff
@@ -144,45 +194,88 @@ def get_bearer_token(app):
         return None
 
 
-def run():
+def update_filings(app: Flask, token: dict):
     """Get filings that haven't been synced with colin and send them to the colin-api."""
-    application = create_app()
+    page = 1
+    limit = 50
+    pending_filings = None
     corps_with_failed_filing = []
+
+    while ((pending_filings is None or page <= math.ceil(pending_filings/limit)) and
+           (results := get_filings(app, token, page, limit))):
+        page += 1
+        pending_filings = results.get('total')
+        if not (filings := results.get('filings')):
+            # pylint: disable=no-member; false positive
+            app.logger.debug('No completed filings to send to colin.')
+        for filing in filings:
+            filing_id = filing['filingId']
+            identifier = filing['filing']['business']['identifier']
+            if identifier in corps_with_failed_filing:
+                # pylint: disable=no-member; false positive
+                app.logger.debug(f'Skipping filing {filing_id} for'
+                                 f' {filing["filing"]["business"]["identifier"]}.')
+            else:
+                colin_ids = send_filing(app, token, filing, filing_id)
+                update = None
+                if colin_ids and any(colin_ids):
+                    update = update_colin_id(app, token, filing_id, colin_ids)
```

Comment:
I found that `colin_ids` can return as `[None]`, which actually evaluates to True. We need to make sure to not attempt updating colin ids if they are None, as it will still insert into the `colin_event_ids` table using a default sequence value.

Reply:
Its not expected to receive `[None]`. When did you receive it? any specific filing?

Reply:
For a successful sync we must receive an event id

Reply:
Yup so I tracked down the source of the issue.

It's caused when an involuntary dissolution filing is passed to colin api without any `meta_data`, or with incorrectly formed `meta_data`. You can see that in [`Filing.add_involuntary_dissolution_event` can return None if this is the case](https://github.com/bcgov/lear/blob/main/colin-api/src/colin_api/models/filing.py#L1178-L1199). This gets [returned as `event_id = None` and saved into the `colinIds`](https://github.com/bcgov/lear/blob/main/colin-api/src/colin_api/resources/filing.py#L166-L174) for the filing (with HTTP status CREATED, even though nothing was created). 

It's probably pretty low risk, as I only encountered it because I made a mistake manually creating local test data.

Thinking more about this, it should probably be fixed in colin api rather than the updater job like I did here. This PR may not get merged (see Argus' comment), but this bug might still be worth fixing anyway.

Reply:
Good catch